en/services/mission.md

```diff
@@ -222,51 +249,157 @@ sequenceDiagram;
 
-### Timeouts and Retries {#timeout}
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: MISSION_REQUEST_PARTIAL_LIST
+    GCS->>Drone: MISSION_REQUEST_INT (start_index)
+    GCS->>GCS: Start timeout
+    Drone-->>GCS: MISSION_ITEM_INT (start_index)
+    Note over GCS,Drone: ... iterate through items ...
+    GCS->>Drone: MISSION_REQUEST_INT (end_index)
+    GCS->>GCS: Start timeout
+    Drone-->>GCS: MISSION_ITEM_INT (end_index)
+    GCS->>Drone: MISSION_ACK
+{% endmermaid %}
 
-All the client (GCS) commands are sent with a timeout.
-If a `MISSION_ACK` is not received before the timeout then the client (GCS) may resend the message.
+In more detail, the sequence of operations is:
+1. GCS sends [MISSION_REQUEST_PARTIAL_LIST](../messages/common.md#MISSION_REQUEST_PARTIAL_LIST) specifying the start and end index values for the required mission items.
+   - No response is expected.
+   - Drone receives the message and prepares for mission item requests.
+1. GCS sends [MISSION_REQUEST_INT](../messages/common.md#MISSION_REQUEST_INT) requesting the first mission item (`seq==start_index`).
+   - A [timeout](#timeout) must be started for the GCS to wait on the response from Drone (`MISSION_ITEM_INT`).
+1. Drone receives `MISSION_REQUEST_INT` and responds with the requested mission item in a [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) message.
+1. GCS and Drone repeat the `MISSION_REQUEST_INT`/`MISSION_ITEM_INT` cycle, iterating `seq` until `end_index` (included) is reached.
+1. After receiving the last mission item the GCS responds with [MISSION_ACK](../messages/common.md#MISSION_ACK) with the `type` of [MAV_MISSION_ACCEPTED](../messages/common.md#MAV_MISSION_ACCEPTED) indicating mission download completion/success.
+   The GCS considers the download complete.
+1. Drone receives `MISSION_ACK` containing `MAV_MISSION_ACCEPTED`and considers the operation to be complete.
+
+
+Note:
+- A [timeout](#timeout) is set for every message that requires a response (e.g. `MISSION_REQUEST_INT`).
+  If the timeout expires without a response being received then the request must be resent.
+- Mission items must be received in order.
+  If an item is received out-of-sequence the expected item should be re-requested by the GCS (the out-of-sequence item is dropped).
+- An [error](#errors) can be signaled in response to any request using a [MISSION_ACK](../messages/common.md#MISSION_ACK) message containing an error code.
+  This must cancel the operation.
+- The sequence above shows the [mission items](#mavlink_commands) packaged in [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) messages.
+  Protocol implementations must also support [MISSION_ITEM](../messages/common.md#MISSION_ITEM) and [MISSION_REQUEST](../messages/common.md#MISSION_REQUEST) in the same way.
+
+
+### Canceling Operations {#cancel}
```

Comment:
@LorenzMeier @WickedShell This is a new section. It is irrelevant to ArduPilot, which does not allow cancellation or treat errors as unrecoverable.
This will need to be updated when https://github.com/mavlink/mavlink/pull/1044 goes in. There will also need to be fixes to PX4/QGC to allow mission cancellation during uploading.